software/firmware/europi.py

```diff
@@ -262,33 +266,32 @@ def __init__(self, pin, debounce_delay=500):
 
     def _bounce_wrapper(self, pin):
         """IRQ handler wrapper for falling and rising edge callback functions."""
-        if self.value() == 1:
+        if self.value() == HIGH:
             if time.ticks_diff(time.ticks_ms(), self.last_rising_ms) < self.debounce_delay:
                 return
             self.last_rising_ms = time.ticks_ms()
             return self._rising_handler()
 
-        elif self.value() == 0:

```

Comment:
[required] I see that you've removed this else as it is logically unnecessary given that the above `if` returns. I agree that that is true, however the `else` block did a nice job of organizing the for the reader. This code for when it is high, and this other code for low. Now that is less clear.

As a remedy I would suggest doing one of the following. Either bring back the else block, or add a one line comment that indicates that the remainder of the method is for the `LOW` case.

Reply:
I agree it's less readable, I'm just trying to improve performances as much as possible but a single else will probably not make much of a difference.
I'll re-add the else.